### PR TITLE
Support for New Relic OTLP endpoint

### PR DESCRIPTION
**Pull request by mdenomy**
Adds support for the [New Relic OTLP](https://docs.fastly.com/en/guides/log-streaming-newrelic-otlp) logging endpoint

### CODE REVIEW

**Comment by mdenomy** [comment]
Ran `make testacc TESTARGS='-run=TestAccFastlyServiceVCL_logging_newrelicotlp_basic'` locally. 

Note that this endpoint should only be used for VCL services, so did not add any support/testing for creating Compute services

File: fastly/block_fastly_service_logging_newrelicotlp.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp.go]: As this is only used for a VCL service, let's remove this `if` statement and inline each of its block attributes into `blockAttributes` above.

File: fastly/block_fastly_service_logging_newrelicotlp.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp.go]: We should remove this as the `pruneVCLLoggingAttribute()` method only works on Compute services...

https://github.com/fastly/terraform-provider-fastly/blob/68c8b8f9aca351d148dfd09fda9c46b2925903aa/fastly/service_attribute_definition.go#L95-L103

File: fastly/block_fastly_service_logging_newrelicotlp.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp.go]: This method is only one line long so we should just inline its implementation here.

Let's also inline `h.buildCreate()` because the redirection is a cognitive flow disruption (IMHO).

File: fastly/block_fastly_service_logging_newrelicotlp.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp.go]: This method is only called here so I would also suggest inlining the implementation as the extra redirection makes it harder to identify what the logic does (if it was called in multiple places I could see a justification for being a separate function, but as it stands I think it just adds more mental/cognitive overhead).

**UPDATE:** Same for `h.buildDelete()` I feel, as it's only a few lines long and the redirection for this makes it harder to reason about the code when the flow is interrupted multiple times.

File: fastly/block_fastly_service_logging_newrelicotlp.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp.go:234]: Is this definitely the case for this API endpoint? I just want to be sure we're not copy/pasting (aka cargo-culting) a pattern that might be relevant only for some other endpoints.

File: fastly/block_fastly_service_logging_newrelicotlp_test.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp_test.go]: nit:

```suggestion
					resource.TestCheckResourceAttr("fastly_service_vcl.foo", "name", name),
					resource.TestCheckResourceAttr("fastly_service_vcl.foo", "logging_newrelicotlp.#", "1"),
```

File: fastly/block_fastly_service_logging_newrelicotlp_test.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp_test.go]: nit:

```suggestion
					resource.TestCheckResourceAttr("fastly_service_vcl.foo", "name", name),
					resource.TestCheckResourceAttr("fastly_service_vcl.foo", "logging_newrelicotlp.#", "2"),
```

File: fastly/block_fastly_service_logging_newrelicotlp_test.go
Comment (Integralist) [inline fastly/block_fastly_service_logging_newrelicotlp_test.go]: This looks like a copy/paste mistake as it's been mentioned that Compute isn't supported for this resource.

```suggestion
```

**Review by Integralist — CHANGES_REQUESTED** [review]
Thanks for re-opening this PR! The CI job passes now which is great 👍🏻 

I have a few requested changes please.

File: fastly/block_fastly_service_logging_newrelicotlp.go
Comment (mdenomy) [reply fastly/block_fastly_service_logging_newrelicotlp.go:234]: Yes, I believe this is true. I made a quick test program using gofastly and tried creating an endpoint with empty strings for both `ResponseCondition` and `Placement` and got 400 errors for each

```
error: 400 - Bad Request:

    Title:  Bad request
    Detail: No condition named ''
    RateLimitRemaining: 994
    RateLimitReset:     1702666800
```

```
error: 400 - Bad Request:

    Title:  Bad request
    Detail: Not allowed placement for ''
    RateLimitRemaining: 993
    RateLimitReset:     1702666800

```

**Review by Integralist — APPROVED** [review]
Thanks @mdenomy for this PR (and your patience waiting for a final review). All looks good to me 👍🏻

**Pull request merged**
